File: dart/lib/model/types_pinning_status.dart
part of swagger.api;

class TypesPinningStatus {
  /// The underlying value of this enum member.
  String value;

  TypesPinningStatus._internal(this.value);

  static TypesPinningStatus pinning_ = TypesPinningStatus._internal("pinning");
  static TypesPinningStatus pinned_ = TypesPinningStatus._internal("pinned");
  static TypesPinningStatus failed_ = TypesPinningStatus._internal("failed");
  static TypesPinningStatus queued_ = TypesPinningStatus._internal("queued");
  static TypesPinningStatus offloaded_ = TypesPinningStatus._internal("offloaded");

  TypesPinningStatus.fromJson(dynamic data) {
    switch (data) {
          case "pinning": value = data; break;
          case "pinned": value = data; break;
          case "failed": value = data; break;
          case "queued": value = data; break;
          case "offloaded": value = data; break;
    default: throw('Unknown enum value to decode: $data');
    }
  }

  static dynamic encode(TypesPinningStatus data) {
    return data.value;
  }
}

File: spring/src/main/java/io/swagger/model/TypesPinningStatus.java
package io.swagger.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonValue;
import org.springframework.validation.annotation.Validated;
import javax.validation.Valid;
import javax.validation.constraints.*;

import com.fasterxml.jackson.annotation.JsonCreator;

/**
 * Gets or Sets types.PinningStatus
 */
public enum TypesPinningStatus {
  PINNING("pinning"),
    PINNED("pinned"),
    FAILED("failed"),
    QUEUED("queued"),
    OFFLOADED("offloaded");

  private String value;

  TypesPinningStatus(String value) {
    this.value = value;
  }

  @Override
  @JsonValue
  public String toString() {
    return String.valueOf(value);
  }

  @JsonCreator
  public static TypesPinningStatus fromValue(String text) {
    for (TypesPinningStatus b : TypesPinningStatus.values()) {
      if (String.valueOf(b.value).equals(text)) {
        return b;
      }
    }
    return null;
  }
}
